packages/authentication/modules/user/useJWTUser/index.ts

```diff
@@ -50,6 +51,12 @@ const useJWTUser = <TUser extends Partial<User> & JWTContent>({
     }
   }, [query.error])
 
+  useEffect(() => {
```

Comment:
I would suggest setting this cooking on the template' middleware ([around here](https://bitbucket.org/silverlogic/baseapp-frontend-template/src/88f345bbeba473356d13435dd05caa111259dfd7/middleware.ts#lines-37)) just to make sure we're always setting this cookie (not relying on `useJWTUser` or similar)

but if feels to me that a language cookie would not be necessary since we have the data available on the authorization cookie already (after decoding it). I think that we could alter `getLanguage` a little bit
```
export const getLanguage = (
  cookieName = ACCESS_COOKIE_NAME,
  { noSSR = false }: ServerSideRenderingOption = {},
) => {
  const user = getUser({ noSSR })
  if (!(user?.preferredLanguage in LanguagesEnum)) return null

  return user?.preferredLanguage
}

```
then on the fetch instance:
```
...
accessCookieName,
...
  const language = getLanguage(accessCookieName, { noSSR: false })
  if (language) {
    fetchOptions.headers!['Accept-Language'] = language
  }

```

Reply:
Hey @anicioalexandre implemented on the middleware :)